dev-docs/RFCs/v6.0/offscreen-render-rfc.md

```diff
@@ -0,0 +1,100 @@
+# RFC: Offscreen Rendering
+
+* **Authors**: Xiaoji
+* **Date**: March 2018
+* **Status**: Pending Review
+
+Notes:
+- Initial implementation of OffscreenAnimationLoop [#453](https://github.com/uber/luma.gl/pull/453)
+- Example usage of OffscreenAnimationLoop [#454](https://github.com/uber/luma.gl/pull/454)
+
+
+## Motivation
+
+WebGL rendering performance can greatly affect UI responsiveness. The experimental [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) API in Chrome and Firefox allows us to move WebGL rendering to a worker thread, and free up the main thread to handle DOM interactions. It is highly desirable for heavy front-end applications.
+
+
+## State of the API
+
+OffscreenCanvas is in active development (behind experimental flags) in both FF and [Chrome](https://www.chromestatus.com/feature/5424182347169792). In Chrome, go to `chrome://flags` and enable "Experimental web platform features."
+
+The API still seems limited/unstable:
+- FF only support OffscreenCanvas with WebGL1 context.
+- Chrome's render buffer does not update in some test scenarios.
+
+
+## Proposal: OffscreenAnimationLoop
+
+Add an `OffscreenAnimationLoop` that mirrors the `AnimationLoop` class but delegate rendering to a worker.
+
+Static methods:
+- `createWorker(opts)` - returns script that can be used to spin up a new Worker. `opts` are the options used to construct an `AnimationLoop` instance on the worker thread.
+
+Constructor:
+- `new OffscreenAnimationLoop(opts)` - `opts` are the same options used to construct an `AnimationLoop`, with additional fields:
+    + `opts.worker` (Worker) - the worker instance running a script created by `OffscreenAnimationLoop.createWorker`
+
+Methods:
+- `setProps(props)` - equivalent to `animationLoop.setProps(props)`
+- `start(opts)` - equivalent to `animationLoop.start(opts)`
+- `stop()` - equivalent to `animationLoop.stop()`
+
+
+### Examples
+
+```
+/// animation.js
+import {GL, Model, Framebuffer, Matrix4} from 'luma.gl';
+
+export default {
```

Comment:
Nit: Would be nice with an interface that avoided the need for such export tricks.

An `AnimationLoop` wrapper class that with an option that resulted in one of these being picked. Understand there is the worker issue...